### PR TITLE
#12243 process documentation for automatic dependency update PRs

### LINKED ISSUE #12243: Create process documentation for automatic dependency update PRs [0.5]
### Problem Description

It's crucial to keep dependencies up to date (security, etc.). Therefore we have introduced automatic PRs for dependencies (#7507). This is a big step forward, but needs to be integrated into the iteration processes to minimize potential follow-up problems of dependency updates.

### Proposed Solution

Document in [CONTRIBUTING.md](https://github.com/SORMAS-Foundation/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) that such automatic updates are to be merged at the start of an iteration (after the release of the previous version) and that the PRs are to "Rebase and Merge" to avoid blown up git history on the development branch.

Include special section for payara and keycloak updates!

### PR DESCRIPTION
**Pull request by MartinWahnschaffe**
Fixes #12243

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12342

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -226,3 +227,26 @@ Once the new version is merged to `master`/`master-<version`, the `release-`/`ho
...
+### Dependency Management
+
+Dependencies are managed in Maven POM files. For most dependencies the version is defined in [sormas-base/pom.xml](sormas-base/pom.xml). \
+[sormas-app](../sormas-app/app/build.gradle) and [sormas-e2e-tests](../sormas-e2e-tests/build.gradle) use Gradle instead.
+
+#### Automatic updates
+
+**Process**: At the beginning of each iteration, look over all [dependency update PRs](https://github.com/SORMAS-Foundation/SORMAS-Project/pulls?q=is%3Apr+is%3Aopen+label%3Adependencies) created by dependabot and process them according to the following rules:
+1. Always use "Rebase and merge" instead of creating a merge commit! If needed use the comment `@dependabot rebase` to rebase the PR to the latest development commit.
+2. If all CI checks are successful, minor versions can be merged
```
Comment (StefanKock) [inline docs/CONTRIBUTING.md:240]: "... minor and micro version updates can be merged."

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -226,3 +227,26 @@ Once the new version is merged to `master`/`master-<version`, the `release-`/`ho
...
+3. For mayor version updates read the changelog to understand implications and possible migration steps of the update. \
```
Comment (StefanKock) [inline docs/CONTRIBUTING.md:241]: "major" (typo)

**Review by StefanKock — CHANGES_REQUESTED** [review]
Some minor text changes

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12342

**Pull request merged**